common/navbar.js

```diff
@@ -6,10 +6,17 @@
         'chaise.authen',
         'ui.bootstrap'
     ])
-    .directive('navbar', ['$window', '$rootScope', 'Session', 'modalUtils', function($window, $rootScope, Session, modalUtils) {
+    .directive('navbar', ['$window', '$rootScope', 'Session', 'modalUtils', 'ConfigUtils', function($window, $rootScope, Session, modalUtils, ConfigUtils) {
+
+        var chaiseConfig = ConfigUtils.getConfigJSON();
 
     // One-time transformation of chaiseConfig.navbarMenu to set the appropriate newTab setting at each node
-        var root = chaiseConfig.navbarMenu = chaiseConfig.navbarMenu || {};
+        var root;
+        if(chaiseConfig.navbarMenu)
+          root = chaiseConfig.navbarMenu;
+        else {
+          root={};
+        }
         if (root) {
```

Comment:
We can even remove this because `root` will always be defined, either as the chaiseConfig property set by the user or to the default value:
```json
{
  "newTab": true,
  "children": []
}
```